toolchain/check/decl_name_stack.cpp

```diff
@@ -276,9 +285,15 @@ static auto CheckQualifierIsResolved(
       CARBON_FATAL("No qualifier to resolve");
 
     case DeclNameStack::NameContext::State::Resolved:
+      if (name_context.resolved_inst_id.is_poisoned()) {
+        context.DiagnoseNameNotFound(name_context.loc_id,
+                                     name_context.unresolved_name_id);
+        return false;
+      }
       return true;
 
     case DeclNameStack::NameContext::State::Unresolved:
+    case DeclNameStack::NameContext::State::Poisoned:
```

Comment:
```suggestion
    case DeclNameStack::NameContext::State::Poisoned:
    case DeclNameStack::NameContext::State::Unresolved:
```

Very minor, but when there's no particular significance to the ordering, I'll suggest keeping things in lexical order.

Reply:
Done.